pkg/watcher/certificate_cache.go

```diff
@@ -0,0 +1,33 @@
+package watcher
+
+import (
+	"time"
+
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/jellydator/ttlcache/v3"
+)
+
+type CertificateCache struct {
+	TTL time.Duration
+	*ttlcache.Cache[string, *certmanagerv1.Certificate]
+}
+
+func NewCertificateCache(ttl time.Duration) *CertificateCache {
+	cache := ttlcache.New[string, *certmanagerv1.Certificate]()
+	go cache.Start()
+	return &CertificateCache{Cache: cache, TTL: ttl}
+}
+
+func (c *CertificateCache) GetCACertFromCache(caCertName string) *certmanagerv1.Certificate {
```

Comment:
A general practice to design in memory cache. It should not return the actual reference to the value saved in the cache, because you make it mutable, which means if in further process, any value of this `Certificate` changed, it directly changed the value in the cache, which is not expected.

That's why always return a copy of the value read from cache, to prevent the content in cache to be changed.

Another suggestion, this cache we only need the `NotBefore` from `Certificate`, it's not necessary to cache whole `Certificate`. @nesmabadr @jeremyharisch